tutorials/chapter1_annexa.md

```diff
@@ -0,0 +1,299 @@
+                                                               0,
+                                                               nullptr,
+                                                               nullptr);
+
+    SimplePublisher simplePublisher(provider.getPublisher());
+
+    provider.getSubscriber()->registerListener("example2",
+                                               SimpleListener<std::string>::staticListener);
+
+    simplePublisher.run();
+
+    provider.getSubscriber()->removeListener("example2");
+}
+```
+
+## Example 3: Passing a class function member which is bound to another one
+
+Functions can be bound to another ones in cases where, for example, the functionality will be the same but the number of arguments differ. In this case, the ```forwarderListenerFunction``` function from the ```SimpleListener``` class has been used:
+
+```cpp
+#include "simple_listener.h"
+#include "simple_publisher.h"
+
+#include <iostream>
+#include <klepsydra/core/event_emitter_middleware_provider.h>
+
+int main()
+{
+    std::cout << "Main thread ID: " << std::this_thread::get_id() << std::endl;
+
+    kpsr::EventEmitterMiddlewareProvider<std::string> provider(nullptr,
+                                                               "tutorial_app_api_example3",
+                                                               0,
+                                                               nullptr,
+                                                               nullptr);
+
+    SimplePublisher simplePublisher(provider.getPublisher());
+    SimpleListener<std::string> simpleListener;
+
+    provider.getSubscriber()->registerListener("example3", simpleListener.forwarderListenerFunction);
+
+    simplePublisher.run();
+
+    provider.getSubscriber()->removeListener("example3");
+}
+```
+
+The relevant control flow of this approach is as follows:
+
+1. The ```SimpleListener``` class is instantiated with the ```simpleListener``` object and its constructor is executed. 
+2. In the ```SimpleListener``` constructor, the ```forwarderListenerFunction``` function is bound to the ```listener``` method. This is achieved by using the ```std::bind``` function with the following parameters:
+  * ```&SimpleListener<T>::listener```: Reference to the ```listener``` method.
+  * ```this```: Pointer to the implicit ```SimpleListener``` object, which is the hidden first parameter of the non-static member functions.
+  * ```std::placeholders::_1```: First argument passed to the ```listener``` method, which is a placeholder that will be replaced by the first argument passed to the ```forwarderListenerFunction``` function.
+3. The listener is registered with the ```forwarderListenerFunction``` function.
+4. When the listener function is called, the ```forwarderListenerFunction``` function acts as a wrapper and calls the ```listener``` method.
+
+## Example 4: Passing a lambda function which captures a variable by const value
```

Comment:
Add sub heading before this `## Lambda function approaches`

Reply:
Also first example can be without any captured variable

Reply:
An example using a lambda function without any captured variable is already shown in [chapter1/src/example1.cpp](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/main/examples/chapter1/src/example1.cpp). I would add in this Annex A other different behaviours a lambda function can have.

Reply:
Oh yes, forgot about that. You can reference that immediately after example 1 here -- the equivalent lambda function to free function is the one given in example 1 of chapter 1.